pkg/auth/authenticator_test.go

```diff
@@ -436,6 +436,61 @@ func TestTestAccessBadIPList(t *testing.T) {
 	assert.Empty(t, extraHeaders, "Expected no extra headers for invalid IP")
 }
 
+func TestTestAccessLimited(t *testing.T) {
+	authenticator := &Authenticator{
+		accessCache:   &AccessCache{},
+		servicesCache: &ServicesCache{},
+	}
+
+	tokenEntry := AccessCacheEntry{
+		AccessToken: cerberusv1alpha1.AccessToken{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "valid-token",
+			},
+			Spec: cerberusv1alpha1.AccessTokenSpec{
+				Priority: 50,
+			},
+		},
+		allowedServices: map[string]struct{}{
+			"SampleWebService": {},
+		},
+	}
+	(*authenticator.accessCache)["valid-token"] = tokenEntry
+
+	// Assuming an IP not in the allow list
+	headers := http.Header{}
+	headers.Set("X-Cerberus-Token", "valid-token")
+
+	request := &Request{
+		Context: map[string]string{
+			"webservice": "SampleWebService",
+		},
+		Request: http.Request{
+			Header: headers,
+		},
+	}
+
+	webservice := ServicesCacheEntry{
+		cerberusv1alpha1.WebService{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "SampleWebService",
+			},
+			Spec: cerberusv1alpha1.WebServiceSpec{
+				LookupHeader:         "X-Cerberus-Token",
+				MinimumTokenPriority: 100,
+			},
+		},
+	}
+
+	(*authenticator.servicesCache)["SampleWebService"] = webservice
+
+	ok, reason, extraHeaders := authenticator.TestAccess(request, webservice)
+
+	assert.False(t, ok, "Expected access to be denied")
+	assert.Equal(t, CerberusReasonAccessLimited, reason, "Expected reason to be AccessLimited")
+	assert.Empty(t, extraHeaders, "Expected no extra headers for AccessLimited")
```

Comment:
@navidnabavi should't we add some header when the access is limited?

Reply:
It wasn't in Pedram's commit and I didn't want to do any changes to the implemented logic. But it may be nice to consider.

Reply:
I think it's not in the scope of this branch, I will create another PR.